### PR TITLE
EWPP-3132: Upgrade library to Symfony 5 and 6.

### PR DESCRIPTION
**Pull request by ademarco**

### CODE REVIEW

**Comment by drupol** [comment]
This is perfect for Symfony 6.

I have some questions:

1. Do you think we could relax some deps to allow Symfony 5 as well? The client who asked this library in first place is using Symfony 5.
2. Just for my own information, what is the benefits of using `~6`  instead of `^6`  in this particular case?

**Comment by ademarco** [comment]
1. I'm actually finalizing it also for Symfony 5.5
2. I don't believe there is any difference when only specifying a major constraint, as in: no minor, no patch

**Comment by drupol** [comment]
Nailed it !

**Comment by drupol** [comment]
Copying the comment from the other thread in here.

About the branches, shouldn't it be a better option to branch off to `1.x` and use the `main` branch for the most recent development ?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -5,26 +5,27 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        PHP_VERSION: ["8.0"]
+        php_version: ["8.1"]
```
Comment (drupol) [inline .github/workflows/ci.yml:8]: Shouldn't we test with `8.0` as well ?

**Comment by ademarco** [comment]
About branches: we prefer having 1.x and 2.x.

**Review by drupol — APPROVED** [review]
:100:

**Comment by drupol** [comment]
> About branches: we prefer having 1.x and 2.x.

Fair enough. That said, I'm curious to know what you're going to have in `main` branch then ?

**Comment by ademarco** [comment]
We will not use `master` at the end, we will just have 1.x and 2.x

**Comment by drupol** [comment]
Ok good to know!

File: composer.json [hunk 2]
```diff
@@ -4,39 +4,41 @@
...
+        "php-http/message": "~1.13",
+        "phpro/soap-client": "~2.4",
         "psr/http-client": "^1",
         "psr/http-client-implementation": "^1",
-        "psr/log": "^1 || ^3",
-        "symfony/serializer": "~4",
-        "symfony/validator": "~4",
-        "veewee/xml": "^1.0"
+        "psr/log": "^3",
+        "symfony/serializer": "~5.5 || ~6",
```
Comment (drupol) [inline composer.json:19]: suggestion
        "symfony/serializer": "~5.4 || ~6",
```

Comment (drupol) [inline composer.json]: Shouldn't it be this ? (there is no Symfony 5.5)

```suggestion
        "symfony/serializer": "~5.4 || ~6",
```

Comment (drupol) [reply composer.json]: Thanks

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ademarco) [reply .github/workflows/ci.yml:8]: We are actually going to keep it on 8.1 only, as library dependencies require that.

**Comment by drupol** [comment]
What's wrong with PHP 8.0 ? Tests were passing fine, no?

**Comment by ademarco** [comment]
Latest version of https://github.com/phpro/soap-client doesn't support it.

File: composer.json [hunk 3]
```diff
@@ -4,39 +4,41 @@
...
+        "php": ">=8.1",
         "ext-curl": "*",
         "ext-dom": "*",
         "ext-soap": "*",
         "ext-xsl": "*",
         "cweagans/composer-patches": "^1",
-        "php-http/logger-plugin": "^1.3",
-        "phpro/soap-client": "~2.0.0",
+        "php-http/logger-plugin": "^1",
+        "php-http/message": "~1.13",
```
Comment (sergepavle) [inline composer.json:14]: Are we really need explicitly mention minimum version dependency? As I understand it prevents failures in tests.
We may mention that in _readme, but it is up to you if it makes sense. As an option, we may leave like that.

**Review by sergepavle — COMMENTED** [review]
Also I've noticed that we have messages during tests executing for --prefer-lowest:
```
Run docker-compose exec -T php ./vendor/bin/phpunit
[21-Feb-2023 10:25:14 UTC] PHP Deprecated:  PHPUnit\Runner\DefaultTestResultCache implements the Serializable interface, which is deprecated. Implement __serialize() and __unserialize() instead (or in addition, if support for old PHP versions is necessary) in /var/www/html/vendor/phpunit/phpunit/src/Runner/DefaultTestResultCache.php on line 34

Deprecated: PHPUnit\Runner\DefaultTestResultCache implements the Serializable interface, which is deprecated. Implement __serialize() and __unserialize() instead (or in addition, if support for old PHP versions is necessary) in /var/www/html/vendor/phpunit/phpunit/src/Runner/DefaultTestResultCache.php on line 34
```

File: composer.json (hunk 3 above)
Comment (ademarco) [reply composer.json:14]: Yeah, I'm afraid we need to do that because of the prefer lowest.

**Review by sergepavle — APPROVED** [review]

**Comment by EC-Hugo-SOMBREIREIRO** [comment]
Hi, 
We're waiting on Symfony 5 / PHP 8 support to be implemented so we can use the client on our project (that needs to go live soon). Any idea when the PRs will be merged?
Thanks.

**Review by upchuk — APPROVED** [review]

**Comment by drupol** [comment]
Great! Thank you guys!

**Pull request merged**